File: P4/civitasruby/lib/civitas/sorpresa_pagar_cobrar.rb
#encoding:utf-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

# Salva

module Civitas
  class Sorpresa_pagar_cobrar < Sorpresa
    def initialize(valor, texto)
      super(texto)
      @valor = valor      
    end
    
    def aplicar_a_jugador(actual,todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos.at(actual).modificar_saldo(@valor)
      end
    end
    
  end
end

File: P5/Civitas/src/civitas/SorpresaPagarCobrar.java
package civitas;
import java.util.ArrayList;
/**
 *
 * @author abel
 */
public class SorpresaPagarCobrar extends Sorpresa {
    
    private float valor;
    
    SorpresaPagarCobrar(int valor, String texto){
        
        this.texto = texto;
        this.valor = valor;       
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}
